jib-core/src/main/java/com/google/cloud/tools/jib/api/JavaContainerBuilder.java

```diff
@@ -127,7 +128,7 @@ public static JavaContainerBuilder from(RegistryImage registryImage) {
   // Keeps track of files to add to the image, by system path
   private final List<PathPredicatePair> addedResources = new ArrayList<>();
   private final List<PathPredicatePair> addedClasses = new ArrayList<>();
-  private final List<Path> addedDependencies = new ArrayList<>();
+  private final List<PathPredicatePair> addedDependencies = new ArrayList<>();
```

Comment:
I think this can be left with `Path`. You use `predicate` only in the new `addDependencies` method and not later.